compiler/rustc_borrowck/src/region_infer/opaque_types.rs

```diff
@@ -239,8 +242,80 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
         // shifting.
         let id_substs = InternalSubsts::identity_for_item(self.tcx, def_id.to_def_id());
         debug!(?id_substs);
-        let map: FxHashMap<GenericArg<'tcx>, GenericArg<'tcx>> =
-            substs.iter().enumerate().map(|(index, subst)| (subst, id_substs[index])).collect();
+
+        let map = substs.iter().enumerate().map(|(index, subst)| (subst, id_substs[index]));
```

Comment:
is this just `iter::zip`?

Reply:
Lol yea, I didn't question the existing code.